scheduler/src/cook/mesos/api.clj

```diff
@@ -1408,15 +1425,23 @@
 (defn check-jobs-and-groups-exist
   "Check if all of the ::jobs and ::groups in the context are valid."
   [conn ctx]
-  (and (check-jobs-exist conn ctx)
-       (check-groups-exist conn ctx)))
+  (let [[jobs-ok? jobs-ctx' :as jobs-result] (vectorize (check-jobs-exist conn ctx))]
+    (if-not jobs-ok?
+      jobs-result
+      (let [[groups-ok? groups-ctx' :as groups-result] (vectorize (check-groups-exist conn ctx))]
+        (cond
+          (not groups-ok?) groups-result
+          (not (or jobs-ctx' groups-ctx')) true
+          (and jobs-ctx' groups-ctx') [true (combine jobs-ctx' groups-ctx')]
+          :else [true (or jobs-ctx' groups-ctx')])))))
 
 (defn validate-retries
   [conn task-constraints ctx]
   (let [retries (get-in ctx [:request :body-params :retries])
         increment (get-in ctx [:request :body-params :increment])
         jobs (jobs-from-request ctx)
         groups (get-in ctx [:request :body-params :groups])
+        failed-only? (get-in ctx [:request :body-params :failed_only])
```

Comment:
Can we make the param use dash instead of underscore? We use dash for other params like `start-ms`, `end-ms` on `/list`

Reply:
I originally had it as `:failed-only?`, but that screwed up my ability to set it via named arguments in our Python integration tests. If you really prefer the hyphen, maybe I can do something in the `retry_jobs` method in Python to convert from snake_case to kabob-case.

Reply:
I thought `cook-coercer` took care of that for us (https://github.com/twosigma/Cook/blob/3e1d88b7747a3c5bf95a2afbd62dec2c269e2c6b/scheduler/src/cook/mesos/api.clj#L1036), but maybe I'm mis-remembering

Reply:
I thought we had something doing that too, but it didn't work when I tried it before. That was probably due to a different schema issue though, so I'll try it again and push the change if it works. Thanks!

Reply:
I didn't realize that I had to update the `cook-coercer` to make this happen. Now it works.